core/src/components/header/header.tsx

```diff
@@ -48,26 +46,30 @@ export class Header implements ComponentInterface {
     this.inheritedAttributes = inheritAttributes(this.el, ['role']);
   }
 
-  async componentDidLoad() {
-    await this.checkCollapsibleHeader();
+  componentDidLoad() {
+    this.checkCollapsibleHeader();
   }
 
-  async componentDidUpdate() {
-    await this.checkCollapsibleHeader();
+  componentDidUpdate() {
+    this.checkCollapsibleHeader();
   }
 
   disconnectedCallback() {
     this.destroyCollapsibleHeader();
   }
 
   private async checkCollapsibleHeader() {
+    const mode = getIonMode(this);
+
+    if (mode !== 'ios') { return; }
+
+    const { collapse } = this;
+    const hasCondense = collapse === 'condense';
+    const hasFade = collapse === 'fade';
+
+    this.destroyCollapsibleHeader();
 
-    // Determine if the header can collapse
-    const hasCollapse = this.collapse === 'condense';
-    const canCollapse = (hasCollapse && getIonMode(this) === 'ios') ? hasCollapse : false;
-    if (!canCollapse && this.collapsibleHeaderInitialized) {
-      this.destroyCollapsibleHeader();
-    } else if (canCollapse && !this.collapsibleHeaderInitialized) {
+    if (hasCondense) {
       const pageEl = this.el.closest('ion-app,ion-page,.ion-page,page-inner');
       const contentEl = (pageEl) ? pageEl.querySelector('ion-content') : null;
```

Comment:
Since we are calling `HTMLIonContentElement.getScrollElement` on this below and in the footer, should we also wait for `componentOnReady` of the `ion-content`? Similar to this PR: https://github.com/ionic-team/ionic-framework/pull/24007